code/modules/tgui/modules/appearance_changer.dm

```diff
@@ -231,30 +231,33 @@
 		data["head_accessory_styles"] = head_accessory_styles
 		data["head_accessory_style"] = head_organ ? head_organ.ha_style : "None"
 
-	data["change_hair"] = can_change(APPEARANCE_HAIR)
-	if(data["change_hair"])
-		var/list/hair_styles = list()
-		for(var/hair_style in valid_hairstyles)
-			hair_styles += list(list("hairstyle" = hair_style))
-		data["hair_styles"] = hair_styles
-		data["hair_style"] = head_organ ? head_organ.h_style : "Skinhead"
-
-	data["change_facial_hair"] = can_change(APPEARANCE_FACIAL_HAIR)
-	if(data["change_facial_hair"])
-		var/list/facial_hair_styles = list()
-		for(var/facial_hair_style in valid_facial_hairstyles)
-			facial_hair_styles += list(list("facialhairstyle" = facial_hair_style))
-		data["facial_hair_styles"] = facial_hair_styles
-		data["facial_hair_style"] = head_organ ? head_organ.f_style : "Shaved"
-
-	data["change_head_markings"] = can_change_markings("head")
-	if(data["change_head_markings"])
-		var/m_style = owner.m_styles["head"]
-		var/list/head_marking_styles = list()
-		for(var/head_marking_style in valid_head_marking_styles)
-			head_marking_styles += list(list("headmarkingstyle" = head_marking_style))
-		data["head_marking_styles"] = head_marking_styles
-		data["head_marking_style"] = m_style
+	if(!isgrey(user) && !ismoth(user) && !isdrask(user))
```

Comment:
Can you make these based on `bodyflags` of the species please?

Reply:
Created a new define for hair gradient as well, as I missed this before. I will resolve this once I make sure it is updated in all the appropriate places.

Reply:
Done!